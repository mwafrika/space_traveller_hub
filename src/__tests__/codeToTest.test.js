/* eslint-disable jsx-quotes */
import React from 'react';
import { createRoot } from 'react-dom/client';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Button from '../codeToTest';

describe('Button', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    createRoot(div).render(<Button />);
  });
  it('it should print the correct button with data', () => {
    render(<Button type='button'>Click me!</Button>);
  });

  it('should match the snapshot', () => {
    const snap = renderer
      .create(<Button type='button'>Welcome</Button>)
      .toJSON();

    expect(snap).toMatchSnapshot();
  });
});
