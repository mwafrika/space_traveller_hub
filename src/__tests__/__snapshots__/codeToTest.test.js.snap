// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<button
  type="button"
>
  Click me!
</button>
`;
